services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java

```diff
@@ -442,6 +442,16 @@ protected Response handleLoginResponse(String samlResponse, SAMLDocumentHolder h
                     assertionElement = DocumentUtil.getElement(holder.getSamlDocument(), new QName(JBossSAMLConstants.ASSERTION.get()));
                 }
 
+                // Validate InResponseTo attribute: must match the generated request ID
+                String expectedRequestId = authSession.getClientNote(SamlProtocol.SAML_REQUEST_ID);
+                final boolean inResponseToValidationSuccess = validateInResponseToAttribute(holder.getSamlDocument().getDocumentElement(), assertionElement, expectedRequestId);
```

Comment:
Can `responseType` be used rather than the raw element from `getDocumentElement()`? It seems that `validateInResponseToAttribute` could leverage that

Reply:
Sure, this makes the checks simpler too. Thanks for spotting this!